### PR TITLE
Rework tooling

### PR DESCRIPTION
**Pull request by CoMPaTech**
Review-note: this (again) reformats our JSONs ... we should decide on how much we want to re-work (or just accept as is)

- [ ] Fixed unreachable LOGGER line in test_init (from pylon)
- [ ] Ignored a few lints (see connected issue, not fixing in this PR)
- [ ] Generating fixtures is optional (as to not regenerate and infinite-loop (again))
- [ ] p1v4 data apparently was not regenerated (started with a fresh fixtures environment
- [ ] we should double-check with #663 but they might run the same circle (couldn't commit without it obviously)
- [ ] leveraging ruff action flow from https://github.com/astral-sh/ruff-pre-commit (using the 'with-fix' approach)

We might need to add another issue for fixing JSON (i.e. maybe not fix it in this already big PR)

## Summary by CodeRabbit

- **New Features**
	- Introduced a new device configuration for a "heater_central_plug" in the JSON structure.
  
- **Bug Fixes**
	- Enhanced error handling in various test cases to ensure proper exception management for connection failures.

- **Documentation**
	- Updated comments and formatting for improved readability across various files.

- **Refactor**
	- Improved code formatting and readability in multiple test files and JSON configurations.
	- Streamlined JSON structures by consolidating multi-line arrays into single-line formats for better readability.

- **Chores**
	- Adjusted configuration settings in `pyproject.toml` for linting and formatting tools.

### CODE REVIEW

File: scripts/tests_and_coverage.sh [hunk 1]
```diff
@@ -35,9 +35,12 @@ if [ -z "${GITHUB_ACTIONS}" ] || [ "$1" == "test_and_coverage" ] ; then
...
 
 if [ -z "${GITHUB_ACTIONS}" ] || [ "$1" == "linting" ] ; then
-    # Black first to ensure nothings roughing up ruff
-    echo "... ruff-ing ..."
+    echo "... biome-ing (prettier) ..."
+    ./tmp/biome check plugwise/ tests/ --files-ignore-unknown=true --no-errors-on-unmatched --indent-width=2 --indent-style=space --write
+    echo "... ruff checking ..."
     ruff check plugwise/ tests/
+    echo "... ruff formatting ..."
+    ruff format plugwise/ tests/
```
[the remark this answers is not in the document]
Comment (CoMPaTech) [reply scripts/tests_and_coverage.sh]: Nice suggestion, worked out a bit differently, hope you like it.

File: tests/data/anna/anna_elga_2.json [hunk 2]
```diff
@@ -37,24 +37,15 @@
     },
     "ebd90df1ab334565b5895f37590ccff4": {
       "active_preset": "home",
-      "available_schedules": [
-        "Thermostat schedule",
-        "off"
-      ],
+      "available_schedules": ["Thermostat schedule", "off"],
```
[the remark this answers is not in the document]
Comment (CoMPaTech) [reply tests/data/anna/anna_elga_2.json:40]: Noted, thought that's actually what we are looking for in this PR to fix, next commit should improve on this

**Comment by CoMPaTech** [comment]
Note: pre-commit.ci disabled for this repo as well

**Comment by CoMPaTech** [comment]
Added additional checks for review in checklist form at the top ^^^

**Review by bouwew — APPROVED** [review]
LGTM! Agreed let's take it step by step.

**Pull request merged**